### PR TITLE
Adding black formatting

### PR DESCRIPTION
**Pull request by Latifah-Umunyana**
# Description
This pull request introduces the integration of the code formatter into our CI/CD pipeline to check the formatting of our Python code during the continuous integration process, ensuring that all code adheres to a consistent style.

#Dependencies
**pre-commit**
sudo pip install pre-commit

Fixes # [(issue) [CI/CD pipeline]](https://lavenderanyango.atlassian.net/browse/TG-60?atlOrigin=eyJpIjoiMTI0YzgzNzVhNjExNGYwZDlkOTk1MzhkMWIxYjBmZmEiLCJwIjoiaiJ9)

## Type of change

- [x] Improvement

# How Has This Been Tested?

- [x] Test A - run: pre-commit run --all-files

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: .github/workflows/taishi.yml [hunk 1]
```diff
@@ -29,3 +29,31 @@ jobs:
...
+
+      - name: Run tests
+        run: |
+          python manage.py test
+
+      - name: Create Pull Request if Black Reformats
+        if: steps.action_black.outcome == 'failure'
+        uses: peter-evans/create-pull-request@v3
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (Ndichu-shee) [inline .github/workflows/taishi.yml:54]: do you have a token?

**Review by cheshari-pearl — APPROVED** [review]

File: .github/workflows/taishi.yml [hunk 2]
```diff
@@ -29,3 +29,29 @@ jobs:
       - name: Run tests
         run: |
           python manage.py test
+
+      - name: Install Black
+        run: |
+          pip install black
```
Comment (Ndichu-shee) [inline .github/workflows/taishi.yml:35]: remove this

File: .github/workflows/taishi.yml [hunk 3]
```diff
@@ -29,3 +29,29 @@ jobs:
...
+
+      - name: Check code formatting with Black
```
Comment (Ndichu-shee) [inline .github/workflows/taishi.yml:37]: remove this

File: .github/workflows/taishi.yml [hunk 4]
```diff
@@ -29,3 +29,20 @@ jobs:
...
+          python manage.py test
+
+      - if: steps.action_black.outcome == 'failure'
+        uses: peter-evans/create-pull-request@v3
+        with:
+          title: "Format Python code with psf/black"
+          commit-message: ":art: Format Python code with psf/black"
+          body: "This pull request formats the code using Black."
+          base: ${{ github.head_ref }}
+          branch: actions/black
```
Comment (Ndichu-shee) [inline .github/workflows/taishi.yml:48]: remove this

**Review by Nancy-Nabacwa — APPROVED** [review]

**Review by aine-mbabazi — APPROVED** [review]

**Review by Ndichu-shee — APPROVED** [review]

**Pull request merged**